### PR TITLE
Dont silently ignore rustdoc errors

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I applied the suggestions from https://github.com/rust-lang/rust/pull/104995 and also checked the rustdoc-ui error but couldn't reproduce it.

r? @notriddle

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
@bors r- failed in a rollup
https://github.com/rust-lang/rust/pull/105444

**Comment by GuillaumeGomez** [comment]
```
src/test/rustdoc-ui/const-evalutation-ice.rs compiled successfully
```

:eyes: Wut?

**Comment by GuillaumeGomez** [comment]
I added an ignore for this platform but that seems very strange it's even needed... Let's run a try to see if it fails on some other platforms too...

@bors try

File: src/test/rustdoc-ui/const-evalutation-ice.rs [hunk 1]
```diff
@@ -8,3 +9,4 @@ pub struct S {
 }
 
 pub const N: usize = 0 - (mem::size_of::<S>() != 4) as usize;
+//~^ ERROR evaluation of constant value failed
```
[the remark this answers is not in the document]
Comment (jyn514) [reply src/test/rustdoc-ui/const-evalutation-ice.rs:12]: Oh, I missed that it was renamed, sorry.

[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/test/rustdoc-ui/const-evalutation-ice.rs:12]: No problem. Like I said I took your changes and just added the comment you mentioned and ran some tests before realizing that your PR was actually super recent. Sorry about that.

[the remark this answers is not in the document]
Comment (jyn514) [reply src/test/rustdoc-ui/const-evalutation-ice.rs:12]: I think the problem is that `S` has different sizes on different platforms, and if the size is 4 then compilation won't fail.

[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/test/rustdoc-ui/const-evalutation-ice.rs:12]: I'll change it to `400`. Just in case.

**Pull request merged**